bin/warehouse/csv_diff.py

```diff
@@ -0,0 +1,107 @@
+import json
+import requests
+import sys
+import csv
+import os
+import codecs
+from sys import stderr, stdout, argv
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+def get_fieldnames(fields):
+    fieldnames = ",".join([f['id'] for f in fields])+",owner_org,log_date,log_activity"
+    return fieldnames
+
+def csv_to_dict(csv_file, primary_keys):
+    csv_dict = {}
+    with open(csv_file) as f:
+        hashcsv = csv.DictReader(codecs.EncodedFile(f, 'utf-8', 'utf-8-sig'), delimiter=",")
+        for row in hashcsv:
+            primary_fields = [str(row[t]) for t in primary_keys]
+            uid = '-'.join(primary_fields)
+            csv_dict[uid] = row
+    return csv_dict
```

Comment:
loading the whole csv into memory is a bad idea as these files will keep getting larger

Reply:
@HoussamBedja this was never addressed. The original script from Andrew loaded data into a DB to avoid this problem but Max removed that work.  Restoring that code might warrant a separate ticket that we can push to the backlog for now.

Reply:
@HoussamBedja @wardi Andrew's code can be found here https://github.com/open-data/ckanext-canada/tree/350b3779e1dc35110110c129ea6ad1b6f70529f8